### PR TITLE
added search_jobs function

### PR DESCRIPTION
**Pull request by kais-viz**

### CODE REVIEW

**Comment by kais-viz** [comment]
This part adds job search function that returns a json object.
Also added it to the docs

File: linkedin_api/linkedin.py [hunk 1]
```diff
@@ -102,7 +103,6 @@ def search(self, params, limit=None, results=[]):
             new_elements.extend(data["data"]["elements"][i]["elements"])
             # not entirely sure what extendedElements generally refers to - keyword search gives back a single job?
             # new_elements.extend(data["data"]["elements"][i]["extendedElements"])
-
```
Comment (tomquirk) [inline linkedin_api/linkedin.py:105]: are these from running black?

File: linkedin_api/linkedin.py [hunk 2]
```diff
@@ -213,9 +208,57 @@ def search_companies(self, keywords=None, limit=None):
...
+
+        # we need to encode the keywords incase it used brackets, otherwise it will return an error
+        if keywords:
+            keywords_encoded = f"&keywords={quote(keywords)}"
+            str_params += keywords_encoded
+        res = self._fetch(
+            f"/search/hits?{str_params}",
+            headers={
+                "accept": "application/vnd.linkedin.normalized+json+2.1",
+                "x-li-track": '{"clientVersion":"1.6.*","osName":"web","timezoneOffset":1,"deviceFormFactor":"DESKTOP","mpName":"voyager-web","displayDensity":1.100000023841858}',
```
Comment (tomquirk) [inline linkedin_api/linkedin.py:255]: **question:** is this essential?

File: tests/test_linkedin_api_requests.py [hunk 3]
```diff
@@ -149,6 +149,12 @@ def test_search_people_by_region(linkedin):
     assert results[0]["public_id"]
 
 
+def test_search_jobs(linkedin):
```
Comment (tomquirk) [inline tests/test_linkedin_api_requests.py:152]: **praise**: for adding this test!

File: linkedin_api/linkedin.py (hunk 1 above)
Comment (kais-viz) [reply linkedin_api/linkedin.py:105]: Yes, I ran it on black playground online. I can rerun it and change some options so it doesn't omit the whitespaces you have in your code

File: linkedin_api/linkedin.py (hunk 2 above)
Comment (kais-viz) [reply linkedin_api/linkedin.py:255]: Yeah it is, I tried running it multiple times without the updated `x-li-track` but it returned the results without actually applying any of the filters apart from the query string and location. In order to add the additional filters it needed the new client version

Comment (tomquirk) [reply linkedin_api/linkedin.py:255]: Very interesting, thanks for checking.  In this case, I'm happy to leave this here. I suspect we'll need to do something similar for the `application/vnd.linkedin.normalized+json+2.1` requests, but no worries for now.

File: linkedin_api/linkedin.py (hunk 1 above)
Comment (tomquirk) [reply linkedin_api/linkedin.py:105]: All good, no worries

File: linkedin_api/linkedin.py [hunk 4]
```diff
@@ -213,9 +208,57 @@ def search_companies(self, keywords=None, limit=None):
                     "subline": item.get("subline", {}).get("text"),
                 }
             )
-
         return results
 
+    def search_jobs(self, keywords, location, count=25, start=0, listedAt=86400):
```
Comment (tomquirk) [inline linkedin_api/linkedin.py:213]: **nitpick**: Can we change `listedAt` to `listed_at`?

Comment (kais-viz) [reply linkedin_api/linkedin.py:213]: yeah ofcourse

Comment (kais-viz) [reply linkedin_api/linkedin.py]: I reverted back the black changes to the file and only applied them to my part of the code, and changed listedAt to listed_at

**Comment by kais-viz** [comment]
I reverted back the black changes to the file and only applied them to my part of the code, and changed listedAt to listed_at. 
Please have a look again!

**Comment by tomquirk** [comment]
@kais-viz thanks for making those changes. LGTM! Thanks again for your contribution! 🎉 ✅ 🙌

**Pull request merged**